packages/@lwc/template-compiler/src/codegen/codegen.ts

```diff
@@ -763,14 +768,21 @@ export default class CodeGen {
                                 )
                             );
                         } else {
-                            // non-class, non-style (i.e. generic attribute or ID/IDRef)
+                            // non-class, non-style (i.e. generic attribute or ID/IDRef or svg use href)
 
                             partToken = `${STATIC_PART_TOKEN_ID.ATTRIBUTE}${partId}:${name}`;
 
                             attributeExpressions.push(
                                 t.property(
                                     t.literal(name),
-                                    bindAttributeExpression(attribute, elm, this, false)
+                                    bindAttributeExpression(
+                                        attribute,
+                                        currentNode,
+                                        this,
+                                        // `addLegacySanitizationHook` is true because `isCustomRendererHookRequired`
+                                        // being false is a precondition for static nodes.
+                                        true
```

Comment:
FYI @jmsjtu I added this bit here. I realized that due to the `isCustomRendererHookRequired` check here, this is always `true`:

https://github.com/salesforce/lwc/blob/275218a0cf5ddaa76f7bbaafef3ec157e429e91e/packages/%40lwc/template-compiler/src/codegen/static-element.ts#L122-L125